### PR TITLE
logthrdest: add syslogng_output_event_retries_total metric 

### PR DESCRIPTION
**Pull request by alltilla**
Example metrics:
```
syslogng_output_event_retries_total{driver="http",url="http://localhost:8888/${path}",id="#anon-destination0#0"} 5
```

### CODE REVIEW

File: lib/logthrdest/logthrdestdrv.c [hunk 1]
```diff
@@ -226,7 +226,13 @@ _connect(LogThreadedDestWorker *self)
                 evt_tag_int("worker_index", self->worker_index),
                 log_expr_node_location_tag(self->owner->super.super.super.expr_node));
       _suspend(self);
+      return;
     }
+
+  if (!self->first_connect)
+    stats_counter_inc(self->owner->metrics.output_event_retries);
```
[the remark this answers is not in the document]
Comment (MrAnno) [reply lib/logthrdest/logthrdestdrv.c:233]: (Around `_rewind_batch()`, or on the error paths of `_process_result()`.)

**Review by MrAnno — CHANGES_REQUESTED** [review]
Reload is a problem here.
With `event_retries`, I think we want to indicate when something went wrong with event delivery.

Multiple `connect` calls are normal within syslog-ng. I think even multiple `disconnect()` calls are normal.
Why don't we increase this counter where actual errors happen during send/flush?

(Around _rewind_batch(), or on the error paths of _process_result().)

**Comment by alltilla** [comment]
Good point. I did not put the counter increase in the error handling functions, because it is only retried after the next connect, which happens after time-reopen, which is 1 minute by default. So we increase the counter 1 minute earlier, but maybe we can live with that. What do you think?

**Comment by MrAnno** [comment]
I think we can live with that :)

**Comment by alltilla** [comment]
Moved it to the result handling functions and it happened to end up around `rewind_batch()`, which is a good indication that it is in the correct place. :) Thanks for the suggestion.

**Comment by MrAnno** [comment]
Thank you. A last question: can't we actually put this into `rewind_batch()`? I see that every call of that function precedes the metric update. `rewind_batch()` is the most direct way of saying "we will retry this" in my opinion, so probably the metric belongs there and remains correct even if we introduce more error paths.

**Comment by alltilla** [comment]
To me it felt a bit side-effecty. If we put it there, I feel like we should rename the function. I will try to come up with something, and if I cannot find, I will just put the stats increase there and not rename it.

**Comment by alltilla** [comment]
@MrAnno Could not come up with a better name :)

**Comment by MrAnno** [comment]
I don't think we need a better name, recording metrics are normal side-effects.

**Review by MrAnno — APPROVED** [review]

**Pull request merged**